task_manager/task_manager/tasks/mission.py

```diff
@@ -81,15 +81,13 @@ def execute_cb(self, goal_handle: ServerGoalHandle) -> MissionAction.Result:
             mission_result.skipped = False
 
             if subtask_result.task_status != TaskStatus.DONE:
-                if subtask.allow_skipping:
+                if subtask.allow_skipping and not goal_handle.is_cancel_requested:
```

Comment:
It took me a while to understand why this change was needed. Even previously before making Missions "cancel_on_stop=True", I would've expecteded that cancelling a Mission will stop the Mission completely, even if subtasks can be skipped.

This didn't seem to be the case previously. When cancelling a whole Mission, we would cancel a subtask, but then actually continue executing the next one if a subtask is allowed to be skipped. Even though the new tasks would be immediatelly cancelled because the Mission goal handle is cancelled, this shouldn't happen because if any of the subtasks were non-cancellable, then the Mission would keep executing.

I'm happy to see that this MR fixes also this issue by not skipping here if the Mission goal handle was cancelled. It would be great to have a test case for this, to confirm that this indeed was a bug previously and that this version now fixes it.

Reply:
Yeah this seemed to be broken to begin with.. sorry I tried to explain the behavior on the description but forgot to mention what did what :smiling_face_with_tear: 

Added a test that verifies this: "mission_canceled" asserts that the goal handle has been canceled... without this addition that assertion fails

Reply:
Nice, thanks! Looks great